### PR TITLE
WIP Lianna/appeals 12530

### PR DESCRIPTION
**Pull request by LiannaNewman**
Resolves #{github issue number}

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Post this PR in #appeals-schema with a summary
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

File: app/controllers/concerns/generic_task_pagination_concern.rb [hunk 1]
```diff
@@ -35,10 +35,14 @@ def paginate_tasks(task_list)
       .per(DEFAULT_TASKS_PER_PAGE)
   end
 
-  def pagination_query_params
+  def unrecognized_column_name_error
+    render json: { error: "Column name provided could not be found" }, status: :not_found
+  end
```
Comment (LiannaNewman) [inline app/controllers/concerns/generic_task_pagination_concern.rb:40]: Remove no longer being used.

Comment (LiannaNewman) [inline app/controllers/concerns/generic_task_pagination_concern.rb:40]:
```suggestion
  def unrecognized_column_name_error
    render json: { error: "Column name provided could not be found" }, status: :not_found
  end
```

File: app/controllers/concerns/generic_task_pagination_concern.rb [hunk 2]
```diff
@@ -35,10 +35,14 @@ def paginate_tasks(task_list)
...
+
+  def pagination_query_params(sort_by_column)
```
Comment (LiannaNewman) [inline app/controllers/concerns/generic_task_pagination_concern.rb:42]:
```suggestion
  def pagination_query_params(sort_by_column: allowed_params[Constants.QUEUE_CONFIG.SORT_COLUMN_REQUEST_PARAM.to_sym])
```

File: client/constants/QUEUE_CONFIG.json [hunk 3]
```diff
@@ -88,11 +88,21 @@
...
     },
+    "CLAIMANT_NAME": {
+      "name": "claimantColumn",
+      "sorting_table": "tasks",
+      "sorting_columns": ["claimant"]
+    },
     "ISSUE_COUNT": {
       "name": "issueCountColumn",
-      "sorting_table": "cached_appeal_attributes",
+      "sorting_table": "tasks",
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:98]:
```suggestion
     "sorting_table": "cached_appeal_attributes",
```

File: client/constants/QUEUE_CONFIG.json [hunk 4]
```diff
@@ -88,11 +88,21 @@
     "BADGES": {
       "name": "badgesColumn"
     },
+    "CLAIMANT_NAME": {
+      "name": "claimantColumn",
+      "sorting_table": "tasks",
+      "sorting_columns": ["claimant"]
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:94]:
```suggestion
     
```

File: client/constants/QUEUE_CONFIG.json [hunk 5]
```diff
@@ -88,11 +88,21 @@
...
+    },
     "ISSUE_COUNT": {
       "name": "issueCountColumn",
-      "sorting_table": "cached_appeal_attributes",
+      "sorting_table": "tasks",
       "sorting_columns": ["issue_count"]
     },
+    "VETERAN_PARTICIPANT_ID": {
+      "name": "veteranParticipantIdColumn",
+      "sorting_table": "tasks",
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:103]:
```suggestion
      
```

File: client/constants/QUEUE_CONFIG.json [hunk 6]
```diff
@@ -88,11 +88,21 @@
...
+      "sorting_columns": ["veteran_participant_id"]
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:104]:
```suggestion
   
```

File: client/test/app/nonComp/NonCompTabs.test.js [hunk 7]
```diff
@@ -1,11 +1,14 @@
 import React from 'react';
```
Comment (LiannaNewman) [inline client/test/app/nonComp/NonCompTabs.test.js:1]: Revert back to original file

File: client/constants/QUEUE_CONFIG.json [hunk 8]
```diff
@@ -88,10 +88,19 @@
     "BADGES": {
       "name": "badgesColumn"
     },
+    "CLAIMANT_NAME": {
+      "name": "claimantColumn",
+
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:93]:
```suggestion
```

File: client/constants/QUEUE_CONFIG.json [hunk 9]
```diff
@@ -88,10 +88,19 @@
...
+    },
     "ISSUE_COUNT": {
       "name": "issueCountColumn",
-      "sorting_table": "cached_appeal_attributes",
+     "sorting_table": "cached_appeal_attributes",
       "sorting_columns": ["issue_count"]
+    },
+    "VETERAN_PARTICIPANT_ID": {
+      "name": "veteranParticipantIdColumn",
+
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:102]:
```suggestion
```

File: client/constants/QUEUE_CONFIG.json [hunk 10]
```diff
@@ -88,10 +88,19 @@
...
     "ISSUE_COUNT": {
       "name": "issueCountColumn",
-      "sorting_table": "cached_appeal_attributes",
+     "sorting_table": "cached_appeal_attributes",
       "sorting_columns": ["issue_count"]
+    },
+    "VETERAN_PARTICIPANT_ID": {
+      "name": "veteranParticipantIdColumn",
+      
+
```
Comment (LiannaNewman) [inline client/constants/QUEUE_CONFIG.json:103]:
```suggestion
```

**Review by ajspotts — APPROVED** [review]
lgtm 🚀

**Pull request merged**